contracts/Swap/SwapRouter.sol

```diff
@@ -717,4 +695,31 @@ contract SwapRouter is Ownable2Step, RouterHelper, IPancakeSwapV2Router {
             revert RepayError(msg.sender, vTokenAddress, response);
         }
     }
+
+    /**
+     * @notice check for min amount out.
+     * @param path The addresses of the underlying token
+     * @param balanceBefore Balance before the swap.
+     * @param amountOutMin Min amount out threshold.
+     */
+    function _checkForAmountOut(
+        address[] calldata path,
+        uint256 balanceBefore,
+        uint256 amountOutMin,
+        address to
+    ) internal view returns (uint256 swapAmount) {
+        uint256 balanceAfter = IERC20(path[path.length - 1]).balanceOf(to);
+        swapAmount = balanceAfter - balanceBefore;
+        require(swapAmount >= amountOutMin, "SwapRouter: SwapAmount is less than amountOutMin");
+    }
+
+    /**
+     * @notice Get swapped amount
```

Comment:
We should try to write better docs. Here something like "returns the difference between the balance of `this` and the balanceBefore" could be better

Reply:
Fixed, 6ef9faf8fc19a066c7e4a7dac440a9bd25f2a1cc